src/source/Dtls/Dtls.c

```diff
@@ -561,6 +569,33 @@ STATUS dtlsSessionPutApplicationData(PDtlsSession pDtlsSession, PBYTE pData, INT
     return retStatus;
 }
 
+STATUS dtlsSessionShutdown(PDtlsSession pDtlsSession)
+{
+    STATUS retStatus = STATUS_SUCCESS;
+    BOOL locked = FALSE;
+
+    CHK(pDtlsSession != NULL, STATUS_NULL_ARG);
+
+    MUTEX_LOCK(pDtlsSession->sslLock);
+    CHK(!ATOMIC_LOAD_BOOL(&pDtlsSession->shutdown), retStatus);
+    CHK(ATOMIC_LOAD_BOOL(&pDtlsSession->sslInitFinished), retStatus);
+
+    SSL_shutdown(pDtlsSession->pSsl);
+    CHK_STATUS(dtlsCheckOutgoingDataBuffer(pDtlsSession));
+
+    ATOMIC_STORE_BOOL(&pDtlsSession->shutdown, TRUE);
+
+    CHK_STATUS(dtlsSessionChangeState(pDtlsSession, CLOSED));
```

Comment:
Are we calling this API with the lock acquired? This is internal so it should be OK